lib/hashes.md

```diff
@@ -5,40 +5,69 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is a Hash?
 
-Replace me with your response...
+It's a place to store data. It's unordered and they use a key/value system instead of number index like arrays. The key must be unique for each hash
 
 # What are some examples of information that would be Hashes as opposed to some other data type?
 
-Replace me with your response...
+You could store the totals number of items a cat likes that another cat also likes. You would be referring to a different table with items that would be stored in a hash 
 
 # How are Hashes and Arrays similar? How are they different?
 
-Replace me with your response...
+They can both store may times of data (integers, floats, strings), but arrays are ordered and hashes are unordered. Hashes are better for easily referencing their values using the key.
 
 # How do you retrieve a particular value from a Hash?
+By references it's particular key
 
-Replace me with your response...
+    clems_friends = Hash.new(0)
+    clems_friends["Kitten"] = 8
+    clems_friends["Gita"] = 6
+    puts clems_friends["Kitten"]
 
+(in repl.it I'm getting the follwing output for the above code and I'm not sure why:
+
+    8
+    => nil
+    
+   I gave the hash a default value (0) so why is it returning 'nil'?)
```

Comment:
That's a common confusion–I agree it's kind of weird. What happening is that the `puts` command displays text and then returns `nil`. That's why the `nil` has the rocket in front of it–it's an indication that the line of code `puts clems_friends["Kitten"]` is complete and that is the result.

`puts` **always** returns `nil` after displaying its text. So your code is correct.